Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -22,43 +22,56 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public object Deserialize(IIonReader reader)
         {
```

Comment:
We should check for argument null here.

Reply:
Done. (also added null check for writer in Serialize())